pandas/core/dtypes/cast.py

```diff
@@ -814,7 +814,7 @@ def infer_dtype_from_scalar(val, pandas_dtype: bool = False) -> tuple[DtypeObj,
             dtype = _dtype_obj
         else:
             dtype = np.dtype("m8[ns]")
-            val = np.timedelta64(val.value, "ns")
+            val = np.timedelta64(val._value, "ns")
```

Comment:
i think this one is expecting the nanos value?  (and probably needs to be updated to allow non-nano?)